python/xir/program.py

```diff
@@ -633,6 +644,138 @@ def merge(*programs) -> "XIRProgram":
 
         return result
 
+    @staticmethod
+    def resolve(library: Mapping[str, "XIRProgram"], name: str) -> "XIRProgram":
+        """Resolves the includes of an XIR program using an XIR library.
+
+        Args:
+            library (Mapping[str, XIRProgram]): mapping from names to XIR programs
+            name (str): name of the root XIR program to resolve
+
+        Returns:
+            XIRProgram: XIR program obtained by recursively merging each XIR
+            program starting from the root XIR program
+
+        **Example**
+
+        Consider an XIR program ``main`` which includes another XIR program ``util``:
+
+        .. code-block:: python
+
+            import xir
+
+            # Create two simple XIR programs.
+            main_program = xir.parse_script("use util; H2 | [0, 1];")
+            util_program = xir.parse_script("gate H2, 0, 2; gate H2: H | [0]; H | [1]; end;")
+
+        The ``main`` XIR program can be resolved using
+
+        .. code-block:: python
+
+            # Define a library containing the XIR programs in the resolution scope.
+            library = {"main": main_program, "util": util_program}
+
+            # Resolve the imports from the main XIR program.
+            resolved_main_program = xir.XIRProgram.resolve(library, "main")
+
+            # Display the result.
+            print(resolved_main_program.serialize())
+
+        The output is
+
+        .. code-block:: none
+
+            gate H2, 0, 2;
+
+            gate H2:
+                H | [0];
+                H | [1];
+            end;
```

Comment:
:thinking: Doesn't the latter gate declaration supersede the former?

Reply:
Ah I think your [answer](https://github.com/XanaduAI/jet/pull/61#discussion_r687097705) cleared things up - I didn't realize there are both declarations and definitions!

Reply:
> Ah I think your [answer](https://github.com/XanaduAI/jet/pull/61#discussion_r687097705) cleared things up - I didn't realize there are both declarations and definitions!

Indeed!

Reply:
Agreed! I've updated the example accordingly.